From 1b0662622aaac9a33b6bf782ba4371530bb1ed9b Mon Sep 17 00:00:00 2001
From: mcontenti <marco.contenti@projectsas.it>
Date: Thu, 22 Aug 2019 18:13:18 +0200
Subject: [PATCH] Location Manager: hack GPS provider check

To use BLE, starting from Android 8, Location Services must me enabled.
Starting from Android 9, a valid Location provider must be available.
This patch let the framework think that a valid provider is available.
---
 .../core/java/com/android/server/LocationManagerService.java    | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/LocationManagerService.java b/services/core/java/com/android/server/LocationManagerService.java
index 180404eef06..93c11b7e59e 100644
--- a/services/core/java/com/android/server/LocationManagerService.java
+++ b/services/core/java/com/android/server/LocationManagerService.java
@@ -2593,7 +2593,7 @@ public class LocationManagerService extends ILocationManager.Stub {
                 for(String provider : mRealProviders.keySet()) {
                     if (provider.equals(LocationManager.PASSIVE_PROVIDER)
                             || provider.equals(LocationManager.FUSED_PROVIDER)) {
-                        continue;
+                        return true;
                     }
                     if (providerList.contains(provider)) {
                         return true;
-- 
2.17.1

